lib/rouge/demos/solidity

```diff
@@ -7,17 +7,6 @@ interface IMirror {
 contract Mirror is IMirror {
     event logMessage(address indexed sender, uint256 value, uint256 gas, bytes data);
 
-    function reflect() external payable returns(bool retval) {
-        assert(msg.sender != address(this));
-        require(msg.value != 0);
-
-        IMirror rorrim = IMirror(msg.sender);
-        retval = rorrim.reflect.value(msg.value).gas(msg.gas)();
-
-        logMessage(msg.sender, msg.value, msg.gas, msg.data);
-        return retval;
-    }
-
```

Comment:
Was it that bad? :(

-----

FTR, there are some parts here (built-in "globals" like `msg.sender`) that should be highlighted; they are listed after

`# TODO: use (currently shadowed by catch-all in :statements)`

in the lexer. (It should probably be highlighted in its entirety.)

But I couldn't get it to work (no good at Ruby), including if moving the `Punctuation` rule in `state :statements` to the very end of the state. Hence the `TODO`.

Reply:
Oh no—the code wasn't bad. The demo is just meant to be a very short example. It's not an issue if it's not syntactically correct; it really isn't visible anywhere except in the list of possible lexers when testing the visual samples with the built-in development server.